config/value.go

```diff
@@ -304,16 +304,30 @@ func convertValue(value interface{}, targetType reflect.Type) (interface{}, erro
 	case string:
 		target := reflect.New(targetType).Interface()
 		switch t := target.(type) {
-		case *int:
-			return strconv.Atoi(v)
-		case *bool:
-			return strconv.ParseBool(v)
+		// case *int:
+		// 	return strconv.Atoi(v)
+		// case *bool:
+		// 	return strconv.ParseBool(v)
 		case *time.Duration:
 			return time.ParseDuration(v)
 		case encoding.TextUnmarshaler:
 			err := t.UnmarshalText([]byte(v))
 			// target should have a pointer receiver to be able to change itself based on text
 			return reflect.ValueOf(target).Elem().Interface(), err
+		default:
```

Comment:
This is going to work only if provider returns a string, not an e.g. int for a Port type, right?